### PR TITLE
[MODULAR] [NO GBP] Donator reward for nikotheguydude

### PR DESCRIPTION
**Pull request by nikothedude**
## About The Pull Request

Re-adds the red victorian dresscoat orion so woefully robbed me off eons ago, in the form of a donator reward for myself.
It has three major things: One, it has a custom desc and examine more text that states its status as clothing of a Tiziran noble (mista). 
Two, it is subtyped off of labcoat/medical. 
Three, it gets shredded if you put it in a washing machine without also throwing in 2.5k credits. This is a in-joke with mista

I slightly edited the unused sprite of the victorian dresscoat to fix a little error in it, where it didnt fully cover the body.

## How This Contributes To The Skyrat Roleplay Experience

i gave you my money

## Proof of Testing

<details>
<summary>Screenshots/Videos</summary>

https://github.com/Skyrat-SS13/Skyrat-tg/assets/59709059/c80783f4-4179-4d60-b09a-720ccd8a7b75

</details>

## Changelog

:cl:
add: The nobility dresscoat, donator reward for NikoTheGuyDude
/:cl:

### CODE REVIEW

File: modular_skyrat/master_files/code/modules/clothing/under/costume.dm
Comment (nikothedude) [inline modular_skyrat/master_files/code/modules/clothing/under/costume.dm:12]: you thought they were gone but you forgot the one crucial aspect: i have 5$

Comment (nikothedude) [reply modular_skyrat/master_files/code/modules/clothing/under/costume.dm:12]: ok it was more like 30$

**Comment by nikothedude** [comment]
shouldve drafted it as i made these changes but im done now, probably

File: modular_skyrat/master_files/code/modules/clothing/under/costume.dm
Comment (ghost) [reply modular_skyrat/master_files/code/modules/clothing/under/costume.dm:12]: Oh, but you fool... I said _dresses_, not the dress-_coats_...

File: modular_skyrat/master_files/code/modules/clothing/under/costume.dm
Comment (ghost) [inline modular_skyrat/master_files/code/modules/clothing/under/costume.dm]: If the base coat isn't available at all in-game, it's best you just make your donator coat a direct subtype of /costume and not bother with re-adding this part. Basically cut out the `icon, worn_icon, icon_state, supports_variations_flags` part of this and slap it onto your item. Better yet, move the icons to the donator .dmis too so they're properly labelled as being paid for

**Comment by ghost** [comment]
Sent a cleaned up icon to match TG standards better, though as I said in the DM you're free to tweak the shading/color. Just stick to the 4-colors and similar shading to TG so it doesn't stick out.

Only other icon-based concern is the Obj icon. Make sure that's set up

**Comment by nikothedude** [comment]
> Sent a cleaned up icon to match TG standards better, though as I said in the DM you're free to tweak the shading/color. Just stick to the 4-colors and similar shading to TG so it doesn't stick out.
> 
> Only other icon-based concern is the Obj icon. Make sure that's set up

bold of you to assume i can sprite in any way whatsoever

File: modular_skyrat/master_files/code/modules/clothing/under/costume.dm
Comment (nikothedude) [reply modular_skyrat/master_files/code/modules/clothing/under/costume.dm]: faaair

**Comment by nikothedude** [comment]
ready

**Comment by nikothedude** [comment]
new video uploaded

File: modular_skyrat/modules/customization/modules/clothing/~donator/donator_clothing.dm
Comment (GoldenAlpharex) [inline modular_skyrat/modules/customization/modules/clothing/~donator/donator_clothing.dm]:
```suggestion
	desc = "An elaborate coat composed of a silky yet firm material. \
		The fabric is quite thin, and provides negligable protection or insulation, \
		but is pleasant on the skin.\n While extremely well made, it seems quite \
		fragile, and rather <i>expensive</i>. You get the feeling it might not \
		<b>survive a washing machine</b> without specialized treatment."
	special_desc = "It's buttons are pressed with some kind of sigil - which, to those knowledgeable in \
		Tiziran politics or nobility, would be recognizable as the <b>Kor'Yesh emblem</b>, \
		a relatively <i>minor house of nobility</i> within <i>Tizira</i>.\n\n\
		It has a strange structure, with many internal clasps, velcro straps, and attachment points. \
		It looks like you could put some other article of clothing into it..."
```

File: modular_skyrat/modules/customization/modules/clothing/~donator/donator_clothing.dm
Comment (GoldenAlpharex) [inline modular_skyrat/modules/customization/modules/clothing/~donator/donator_clothing.dm]: Does it really have to be a list if it's just for one type? Doesn't really feel necessary to me.

File: modular_skyrat/modules/customization/modules/clothing/~donator/donator_clothing.dm
Comment (Iamgoofball) [inline modular_skyrat/modules/customization/modules/clothing/~donator/donator_clothing.dm]: under no circumstances should donators be getting items that let them conceal armor in the visual of a non-armor clothing item

in fact, why is this not just a subtype of labcoat? what does this entire janky absorb thing serve other than letting this thing pretend to be a labcoat without being a labcoat?

File: modular_skyrat/modules/customization/modules/clothing/~donator/donator_clothing.dm
Comment (Iamgoofball) [inline modular_skyrat/modules/customization/modules/clothing/~donator/donator_clothing.dm:1679]: needs to have digitigrade variants, this is enforced policy per spritetainers last i was informed

**Comment by Iamgoofball** [comment]
also, you need to have a changelog, we changelog all changes

File: modular_skyrat/modules/customization/modules/clothing/~donator/donator_clothing.dm
Comment (nikothedude) [reply modular_skyrat/modules/customization/modules/clothing/~donator/donator_clothing.dm:1679]: please watch the video, the clothing CANT have digi variants because it doesnt interact with legs at all

**Comment by nikothedude** [comment]
> also, you need to have a changelog, we changelog all changes

Seriously? who the hell is going to care about _my_ donator item? someone reading the CL and seeing "niko has a donator item" really is not going to give a _shit_
goof i respect you but i'll defer to a maintainer on this

File: modular_skyrat/modules/customization/modules/clothing/~donator/donator_clothing.dm
Comment (nikothedude) [reply modular_skyrat/modules/customization/modules/clothing/~donator/donator_clothing.dm]: I originally made this with the intent of having it be able to mimic multiple types of inconsequential clothing, but quickly realized the labcoat was the only _truely_ inconsequential one I wanted to mimic

Winter coats' behavior are defined by their sprite, armor vests are a good old _fuck no_, and everything else doesn't really matter

Comment (nikothedude) [reply modular_skyrat/modules/customization/modules/clothing/~donator/donator_clothing.dm]: I COULD cut it all out and just subtype this onto labcoat, but 
1. The dresscoat simply is _not_ a labcoat, it's fabric is thin and basically only there to make a thin-ass dress
2. there may, in fact, come a day i realize there is another thing i want to wear under this

its a fair point though and i'll thinka bout it

**Review by GoldenAlpharex — CHANGES_REQUESTED** [review]
I forgot to actually send it.

File: modular_skyrat/modules/customization/modules/clothing/~donator/donator_clothing.dm
Comment (nikothedude) [reply modular_skyrat/modules/customization/modules/clothing/~donator/donator_clothing.dm]: yeah i can just remove this for no no biggie

Comment (nikothedude) [reply modular_skyrat/modules/customization/modules/clothing/~donator/donator_clothing.dm]: ill test this real quick

File: modular_skyrat/modules/customization/modules/clothing/~donator/donator_clothing.dm
Comment (vinylspiders) [inline modular_skyrat/modules/customization/modules/clothing/~donator/donator_clothing.dm]:
```suggestion
		The fabric is quite thin, and provides negligible protection or insulation, \
```

**Review by vinylspiders — CHANGES_REQUESTED** [review]

File: modular_skyrat/modules/customization/modules/clothing/~donator/donator_clothing.dm
Comment (ghost) [reply modular_skyrat/modules/customization/modules/clothing/~donator/donator_clothing.dm:1679]: This is false anyways, Donator clothing explicitly does not need to conform to this. It's intended for use by the donator and people of their choosing, and if they don't choose digitigrade (or even if they ONLY choose digitigrade) that's on them.
Plus this is a jacket goof. A suit jacket. That doesn't cover legs.

**Review by vinylspiders — APPROVED** [review]

**Review by GoldenAlpharex — APPROVED** [review]
Looks much nicer now, thank you Niko.

**Pull request merged**
